processing/src/main/java/org/apache/druid/collections/spatial/search/RadiusBound.java

```diff
@@ -23,37 +23,60 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Predicate;
 import com.google.common.collect.Iterables;
-import org.apache.druid.collections.spatial.ImmutablePoint;
+import org.apache.druid.collections.spatial.ImmutableFloatPoint;
+import org.apache.druid.collections.spatial.RTreeUtils;
 
 import java.nio.ByteBuffer;
 
 /**
+ *
  */
 public class RadiusBound extends RectangularBound
 {
   private static final byte CACHE_TYPE_ID = 0x01;
   private final float[] coords;
   private final float radius;
+  private final RadiusUnit radiusUnit;
 
   @JsonCreator
   public RadiusBound(
       @JsonProperty("coords") float[] coords,
       @JsonProperty("radius") float radius,
-      @JsonProperty("limit") int limit
+      @JsonProperty("limit") int limit,
+      @JsonProperty("radiusUnit") RadiusUnit radiusUnit
```

Comment:
should probably add a serde test to `RadiusBoundTest` with the new parameter. also it should be marked `@Nullable`

Reply:
added more tests for nullable